server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSource.java

```diff
@@ -363,17 +437,42 @@ public SortedBinaryDocValues bytesValues(LeafReaderContext context) throws IOExc
 
         };
 
-        /** Whether the underlying data is floating-point or not. */
+        /**
+         * Are values of this field better represented as a double precision
+         * floating point numbers ({@code true}) or 64 bit signed
+         * numbers ({@code false})?
+         * <p>
+         * Aggregations may, if they feel it is important, use this to pick
+         * which of {@link #longValues} and {@link #doubleValues} is better for
+         * the field values. Most metric aggregations are quite happy to operate
+         * on floating point numbers all the time and never call this. Bucketing
+         * aggregations that want to enumerate all values
+         * (like {@link TermsAggregator}) will want to check this but bucketing
+         * aggregations that just compare values ({@link RangeAggregator}) are,
+         * like metric aggregators, fine ignoring it.
+         */
         public abstract boolean isFloatingPoint();
 
-        /** Get the current {@link SortedNumericDocValues}. */
+        /**
+         * Get a 64 bit signed view into the values in this leaf.
+         * <p>
+         * If the values have precision beyond the decimal point then they'll
+         * be rounded but they'll accurately represent values up
```

Comment:
Do we actually round when casting from double to long?  I thought we just dropped the fractional part.

Reply:
Dropping the fractional part is a kind of rounding. I imagine we just do a java cast.

Reply:
yeah, but usually folks read "rounding" to mean "bank rounding", e.g. down if under 5, up otherwise.